common/primitives/src/messages.rs

```diff
@@ -13,15 +13,18 @@ use utils::*;
 /// A type for responding with an single Message in an RPC-call.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
-pub struct MessageResponse<AccountId, BlockNumber> {
+pub struct MessageResponse<BlockNumber> {
 	#[cfg_attr(feature = "std", serde(with = "as_hex"))]
 	/// Serialized data in a the schemas.
 	pub payload: Vec<u8>,
-	/// The public key of the provider and the signer of the transaction.
-	pub provider_key: AccountId,
-	/// Message source account id (the original source).
-	pub msa_id: MessageSourceId,
-	/// Index in block to get total order
+	/// Message source account id of the MSA that signed the transaction. This may be the same id
+	/// as contained in `original_msa_id`, indicating that the transaction to add the message was
+	/// signed by the originator. An id differing from that of `original_msa_id` indicates that a
+	/// provider was delegated by `original_msa_id` to send this message on its behalf.
+	pub signer_msa_id: MessageSourceId,
+	///  Message source account id (the original source).
+	pub original_msa_id: MessageSourceId,
```

Comment:
This could also be left alone as `msa_id`. So long as documentation indicates that this field represents the MSA that the message is from, regardless if a provider signed the transaction

Reply:
Looks good to me.